src/System.Drawing.Common/tests/mono/System.Imaging/MetafileTest.cs

```diff
@@ -0,0 +1,705 @@
+//
+// Copyright (C) 2007 Novell, Inc (http://www.novell.com)
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+using System;
+using System.Drawing;
+using System.Drawing.Drawing2D;
+using System.Drawing.Imaging;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Security.Permissions;
+using Xunit;
+
+namespace MonoTests.System.Drawing.Imaging
+{
+
+    public class MetafileTest
+    {
+
+        public const string Bitmap = "non-inverted.bmp";
+        public const string WmfPlaceable = "telescope_01.wmf";
+        public const string Emf = "milkmateya01.emf";
+
+        [ConditionalFact(Helpers.GdiplusIsAvailable)]
+        public void Metafile_Stream_Null()
+        {
+            AssertExtensions.Throws<ArgumentNullException, ArgumentException>("stream", null, () => new Metafile((Stream)null));
+        }
+
+        [ConditionalFact(Helpers.GdiplusIsAvailable)]
+        public void Metafile_String_Null()
+        {
+            Assert.Throws<ArgumentNullException>(() => new Metafile((string)null));
+        }
+
+        [ConditionalFact(Helpers.GdiplusIsAvailable)]
+        public void Metafile_String_Empty()
+        {
+            Assert.Throws<ArgumentException>(() => new Metafile(String.Empty));
+        }
+
+        [ConditionalFact(Helpers.GdiplusIsAvailable)]
+        public void Metafile_String_FileDoesNotExists()
+        {
+            string filename = Helpers.GetTestBitmapPath("telescope_02.wmf");
+            Assert.Throws<ExternalException>(() => new Metafile(filename));
+        }
+
+        [ConditionalFact(Helpers.GdiplusIsAvailable)]
+        public void Metafile_String()
+        {
+            string filename = Helpers.GetTestBitmapPath(WmfPlaceable);
+            Metafile mf = new Metafile(filename);
+            Metafile clone = (Metafile)mf.Clone();
+        }
+
+        [ConditionalFact(Helpers.GdiplusIsAvailable)]
+        public void GetMetafileHeader_Bitmap()
```

Comment:
Duplicate of Ctor_Bitmap_ThrowsExternalException -- if this is passing then we can enable that other one.